pkgs/development/ocaml-modules/synchronizer/default.nix

```
@@ -0,0 +1,32 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildDunePackage,
+  prelude,
+}:
+
+buildDunePackage rec {
+  pname = "synchronizer";
+  version = "0.1";
+
+  minimalOCamlVersion = "4.14";
+
+  src = fetchFromGitHub {
+    owner = "OCamlPro";
+    repo = pname;
+    tag = version;
+    hash = "sha256-VlKqORXTXafT88GXHIYkz+A1VkEL3jP9SMqDdMyEdrw=";
+  };
+
+  propagatedBuildInputs = [
+    prelude
+  ];
+
+  meta = {
+    homepage = "https://github.com/OCamlPro/synchronizer";
+    description = "A synchronizer to make datastructures thread-safe.";
+    changelog = "https://raw.githubusercontent.com/OCamlPro/synchronizer/refs/tags/${src.tag}/CHANGES.md";
```

```suggestion
    changelog = "https://raw.githubusercontent.com/OCamlPro/synchronizer/${src.rev}/CHANGES.md";
```
this way the link will generate correctly even if you set up an unstable commit in the source